File: leetcode/lc-us/cpp/2518.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    int countPartitions(vector<int> &nums, int k) {
        const int MOD = 1e9 + 7;
        if (accumulate(nums.begin(), nums.end(), 0LL) < 2 * k) return 0;
        int f[k];
        memset(f, 0, sizeof f);
        f[0] = 1;
        int res = 1;
        for (int x: nums) {
            res = res * 2 % MOD;
            for (int j = k - 1; j >= x; j--) {
                f[j] = (f[j] + f[j - x]) % MOD;
            }
        }
        for (int x: f) {
            res = (res - x * 2 % MOD + MOD) % MOD;
        }
        return res;
    }
};

File: leetcode/lc-us/py/2518.py
from typing import List


class Solution:
    def countPartitions(self, nums: List[int], k: int) -> int:
        if sum(nums) < k * 2:
            return 0
        mod = 10 ** 9 + 7
        tot = pow(2, len(nums), mod)
        f = [1] + [0] * (k - 1)
        for x in nums:
            for i in range(k - 1, x - 1, -1):
                f[i] = (f[i] + f[i - x]) % mod
        return ((tot - sum(f) * 2) % mod + mod) % mod
